Get for ConfigMap

exarkun opened the pull request:
Fixes #15

exarkun committed:
Add an integration test for getting one ConfigMap.

Also refactor the similar test for Namespace - before I remembered the
cases aren't quite compatible due to global vs namespaced difference.

exarkun committed:
Add ConfigMap.named to simplify creation for .get()

exarkun committed:
Implement single ConfigMap retrieval for the fake

rodrigc commented on src/txkube/_memory.py:
```diff
@@ -132,7 +137,8 @@ def _delete(self, request, collection, collection_name, name):
     def not_found(self, request, name):
-        request.responseHeaders.setRawHeader(u"content-type", [u"application/json"])
+        # XXX Untested - https://github.com/LeastAuthority/txkube/issues/42
+        request.responseHeaders.setRawHeaders(u"content-type", [u"application/json"])
```
Adding a test here as specified in https://github.com/LeastAuthority/txkube/issues/42 would be nice

exarkun replied:
For sure, eventually.  Filed #42 so I could skip it for now. :)

rodrigc commented on src/txkube/_memory.py:
```diff
@@ -100,18 +100,23 @@ class _Kubernetes(object):
-                u"items",
-                pset(obj for obj in collection.items if obj.metadata.namespace == namespace),
-            )
+            collection = self._reduce_to_namespace(collection, namespace)
```
Might be nice to add a test for this.

rodrigc reviewed (approved):
Looks OK, more tests can be added later.

exarkun committed:
Merge remote-tracking branch 'origin/master' into 15.get-configmap

exarkun committed:
Test that getting an object respects namespace

Partially implement the proper 404 that results from the way the test is
written.  Delay most of that work for #42.
